src/state/sagas/windows.js

```diff
@@ -43,7 +43,7 @@ export function* setWindowStartingCanvas(action) {
   const windowId = action.id || action.window.id;
 
   if (canvasId) {
-    const thunk = yield call(setCanvas, windowId, canvasId);
+    const thunk = yield call(setCanvas, windowId, canvasId, null, { preserveViewport: true });
```

Comment:
```suggestion
    const thunk = yield call(setCanvas, windowId, canvasId, null, { preserveViewport: !!action.payload });
```

I think we'll also want to reset the viewport if this was an update (because we get here if the manifest changes?)